src/OptionsCompounderAave2.sol

```diff
@@ -22,92 +26,56 @@ error OptionsCompounder__FlashloanNotProfitable(
     uint256 fundsAvailable,
     uint256 fundsToPay
 );
+error OptionsCompounder__AssetNotEqualToPaymentToken();
+error OptionsCompounder__NotFinished();
+
+address constant BEETX_VAULT_OP = 0xBA12222222228d8Ba445958a75a0704d566BF2C8;
 
 /* Main contract */
-contract OptionsCompounder is FlashLoanReceiverBase, Ownable {
+// TODO: UUPSUpgradeable ?
+contract OptionsCompounder is IFlashLoanReceiver {
     /* Constants */
     uint8 constant MIN_NR_OF_FLASHLOAN_ASSETS = 1;
-    address constant BEETX_VAULT_OP =
-        0xBA12222222228d8Ba445958a75a0704d566BF2C8;
 
     /* Storages */
     IOptionsToken private optionToken;
+    ILendingPoolAddressesProvider private addressProvider;
     ISwapperSwaps private swapperSwaps;
-    address private vaultAddress; // BEETx vault not a strategy vault
-    address[] private strategies;
-    uint256 gain = 0;
-
-    address lastExecutor;
-    bool flashloanFinished = true;
-    mapping(address => uint256) senderToBalance; // storage variable which tracks amount to withdraw by address
-
-    modifier onlyStrategy() {
-        if (false == isStrategyAdded(msg.sender)) {
-            revert OptionsCompounder__NotAStrategy();
-        }
-        _;
-    }
+    ILendingPool private lendingPool;
+    address wantToken;
+    bool flashloanFinished;
+    uint256 gain = 0; // TODO: remove at the end
 
     /**
      * List of params which are initiated at the begining:
      * @param _optionToken - option token address which allows to redeem underlying token via operation "exercise"
      * @param _addressProvider - address lending pool address provider - necessary for flashloan operations
      * @param _reaperSwapper - address to contract allowing to swap tokens in easy way
-     * @param _strategies - strategies which can use this module
      * */
-    constructor(
+    // TODO: Add onlyInitializing !
+    function _OptionsCompounder_Init(
```

Comment:
convention is double wide __ here im pretty sure

Reply:
also ya, make sure to add onlyInitializing

Reply:
```
    /// @custom:oz-upgrades-unsafe-allow constructor
    constructor() initializer {}
```

Reply:
u might need to add this too

Reply:
https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#initializing_the_implementation_contract